client/webserver/webserver_test.go

```diff
@@ -610,3 +620,63 @@ func TestGetOrderIDCtx(t *testing.T) {
 		}
 	}
 }
+
+func TestPasswordCache(t *testing.T) {
+	s, tCore, shutdown, err := newTServer(t, false)
+	if err != nil {
+		t.Fatalf("error starting server: %v", err)
+	}
+	defer shutdown()
+
+	password := encode.PassBytes("def")
+	authToken1 := s.authorize()
+	authToken2 := s.authorize()
+
+	key1, err := s.cacheAppPassword(password, authToken1)
+	if err != nil {
+		t.Fatalf("error caching password: %v", err)
+	}
+
+	key2, err := s.cacheAppPassword(password, authToken2)
+	if err != nil {
+		t.Fatalf("error caching password: %v", err)
+	}
+
+	retrievedPW, err := s.getCachedPassword(key1, authToken1)
+	if err != nil {
+		t.Fatalf("error getting password: %v", err)
+	}
+	if !bytes.Equal(password, retrievedPW) {
+		t.Fatalf("retrieved PW not same: %v - %v", password, retrievedPW)
+	}
+
+	retrievedPW, err = s.getCachedPassword(key2, authToken2)
+	if err != nil {
+		t.Fatalf("error getting password: %v", err)
+	}
+	if !bytes.Equal(password, retrievedPW) {
+		t.Fatalf("retrieved PW not same: %v - %v", password, retrievedPW)
+	}
+
+	// test new wallet request first without the cookies populated, then with
+	writer := new(TWriter)
+	reader := new(TReader)
+	body := &newWalletForm{
+		Pass: encode.PassBytes(""),
+	}
+	want := `{"ok":false,"msg":"password error: app pass cannot be empty"}`
+	tCore.notHas = true
+	ensureResponse(t, s.apiNewWallet, want, reader, writer, body, nil)
+
+	want = `{"ok":true}`
+	ensureResponse(t, s.apiNewWallet, want, reader, writer, body, map[string]string{
+		authCK:  authToken1,
+		pwKeyCK: hex.EncodeToString(key1),
```

Comment:
We should store the serialized `Crypter` locally and only send the key in the cookie. How about we update the `authTokens map[string]bool` field to be something like

```Go
auths map[string]*struct{
    CachedCrypter encrypt.Crypter
}
```

then if they provide a key we can regenerate, but nobody ever has all the pieces needed in memory.

Reply:
That makes sense, then the cookie can be smaller and we don't have to regenerate the crypter each time.
How about having all the password related things in a cachedPasswords map like this:
```go
cachedPasswords map[string]*struct{
    EncryptedPass []byte
    Crypter encrypt.Crypter 
}

Reply:
If you have the `Crypter` then you don't need the key anymore. I ended up caching the serialized crypter instead.